lib/signaling/v2/peerconnection.js

```diff
@@ -1129,12 +1131,25 @@ class PeerConnectionV2 extends StateMachine {
         description.sdp = enableDtxForOpus(description.sdp, []);
       }
 
-      // NOTE(mroberts): Do this to reduce our MediaStream count in Firefox. By
-      // mapping MediaStream IDs in the SDP to "-", we ensure the "track" event
-      // doesn't include any new MediaStreams in Firefox. Its `streams` member
-      // will always be the empty Array.
       if (isFirefox) {
+        // NOTE(mroberts): Do this to reduce our MediaStream count in Firefox. By
+        // mapping MediaStream IDs in the SDP to "-", we ensure the "track" event
+        // doesn't include any new MediaStreams in Firefox. Its `streams` member
+        // will always be the empty Array.
         description.sdp = filterOutMediaStreamIds(description.sdp);
+
+        // NOTE(mmalavalli): Re-ordering payload types in order to make sure a non-H264
+        // preferred codec is selected does not work on Mobile Firefox due to this behavior:
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=1683258. So, we remove H264 from the
+        // remote sdp in order to fix VIDEO-3736.
+        const preferredVideoCodec = this._preferredVideoCodecs[0];
+        const peerSdp = this._peerConnection.localDescription
+          ? this._peerConnection.localDescription.sdp
+          : description.sdp;
+        if (preferredVideoCodec && preferredVideoCodec.codec.toLowerCase() !== 'h264') {
+          description.sdp = unifiedPlanFilterRemoteVideoCodecs(description.sdp, peerSdp, ['h264']);
+        }
```

Comment:
This is the main fix: Mobile Firefox prefers H264 even if a non-H264 codec is preferred. So, we need to remove H264 from the remote SDP so that Mobile Firefox is forced to use VP8. One caveat: if the remote m= section has only H264, then we should not remove it since it does not support VP8.